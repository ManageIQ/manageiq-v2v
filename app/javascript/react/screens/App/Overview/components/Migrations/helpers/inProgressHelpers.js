import getMostRecentVMTasksFromRequests from './getMostRecentVMTasksFromRequests';
import getMostRecentRequest from '../../../../common/getMostRecentRequest';
import { PLAN_JOB_STATES } from '../../../../../../../data/models/plans';
import { urlBuilder } from '../helpers';
import { TRANSFORMATION_PLAN_REQUESTS_URL } from '../../../OverviewConstants';

export const getRequestsOfPlan = (plan, allRequestsWithTasks) => {
  const requestsOfAssociatedPlan = allRequestsWithTasks.filter(request => request.source_id === plan.id);
  const mostRecentRequest = requestsOfAssociatedPlan.length > 0 && getMostRecentRequest(requestsOfAssociatedPlan);
  return { requestsOfAssociatedPlan, mostRecentRequest };
};

export const isWaitingForConversionHost = mostRecentRequest =>
  mostRecentRequest &&
  mostRecentRequest.approval_state === 'approved' &&
  mostRecentRequest.miq_request_tasks.length > 0 &&
  mostRecentRequest.miq_request_tasks.every(task => !task.conversion_host_id);

export const shouldDisableCancelButton = ({
  requestsProcessingCancellation,
  mostRecentRequest,
  isFetchingTransformationPlans,
  isFetchingAllRequestsWithTasks,
  isCancellingPlanRequest
}) => {
  const isProcessingCancellation = requestsProcessingCancellation.includes(
    urlBuilder(TRANSFORMATION_PLAN_REQUESTS_URL, mostRecentRequest.id)
  );

  return (
    isProcessingCancellation &&
    (isFetchingTransformationPlans ||
      isFetchingAllRequestsWithTasks ||
      isCancellingPlanRequest ||
      !!mostRecentRequest.cancelation_status)
  );
};

export const countFailedVms = mostRecentRequest => {
  let failed = false;
  let numFailedVms = 0;
  mostRecentRequest.miq_request_tasks.forEach(task => {
    if (task.status === 'Error') {
      failed = true;
      numFailedVms += 1;
    }
  });
  return { failed, numFailedVms };
};

export const getIndexedTasksOfPlan = (plan, requestsOfAssociatedPlan, mostRecentRequest) => {
  const tasksBySourceId = {};
  let tasksOfPlan = {};
  if (requestsOfAssociatedPlan.length > 0) {
    tasksOfPlan = getMostRecentVMTasksFromRequests(requestsOfAssociatedPlan, plan.options.config_info.actions);
  } else if (mostRecentRequest) {
    tasksOfPlan = mostRecentRequest.miq_request_tasks;
  }
  tasksOfPlan.forEach(task => {
    tasksBySourceId[task.source_id] = tasksBySourceId[task.source_id] || {};
    tasksBySourceId[task.source_id].completed = task.status === 'Ok' && task.state === 'finished';
    tasksBySourceId[task.source_id].virtv2v_disks = task.options.virtv2v_disks;
    tasksBySourceId[task.source_id].playbooks = task.options.playbooks;
  });
  return tasksBySourceId;
};

const isPlaybookRunning = playbook =>
  playbook && (playbook.job_state === PLAN_JOB_STATES.PENDING || playbook.job_state === PLAN_JOB_STATES.ACTIVE);

export const aggregateTasks = tasksBySourceId => {
  const sourceIds = Object.keys(tasksBySourceId);
  const numTotalVms = sourceIds.length;
  let numCompletedVms = 0;
  let taskRunningPreMigrationPlaybook = null;
  let taskRunningPostMigrationPlaybook = null;
  sourceIds.forEach(sourceId => {
    const task = tasksBySourceId[sourceId];
    if (task.completed) numCompletedVms += 1;
    // after we have assigned the latest tasks for each source, check the running playbook status
    if (task.playbooks) {
      if (isPlaybookRunning(task.playbooks.pre)) {
        taskRunningPreMigrationPlaybook = task;
      }
      if (isPlaybookRunning(task.playbooks.post)) {
        taskRunningPostMigrationPlaybook = task;
      }
    }
  });
  return { numTotalVms, numCompletedVms, taskRunningPreMigrationPlaybook, taskRunningPostMigrationPlaybook };
};

export const calculateTotalDiskSpace = tasksBySourceId => {
  let totalDiskSpace = 0;
  let totalMigratedDiskSpace = 0;
  Object.keys(tasksBySourceId).forEach(sourceId => {
    const taskDisks = tasksBySourceId[sourceId].virtv2v_disks;
    if (taskDisks && taskDisks.length) {
      const totalTaskDiskSpace = taskDisks.reduce((a, b) => a + b.size, 0);
      const percentComplete = taskDisks.reduce((a, b) => a + b.percent, 0) / (100 * taskDisks.length);
      const taskDiskSpaceCompleted = percentComplete * totalTaskDiskSpace;

      totalDiskSpace += totalTaskDiskSpace;
      totalMigratedDiskSpace += taskDiskSpaceCompleted;
    }
  });
  return { totalDiskSpace, totalMigratedDiskSpace };
};
