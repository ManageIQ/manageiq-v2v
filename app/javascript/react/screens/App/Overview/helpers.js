export const planRequestWithPlanId = (
  transformationPlanRequest,
  transformationPlans
) => {
  const matchingPlan = transformationPlans.find(transformationPlan =>
    transformationPlan.miq_requests.some(
      request => request.id === transformationPlanRequest.id
    )
  );

  return matchingPlan
    ? { ...transformationPlanRequest, plan_id: matchingPlan.id }
    : transformationPlanRequest;
};

export const planRequestsWithPlanIds = (
  transformationPlanRequests,
  transformationPlans
) =>
  transformationPlanRequests.map(transformationPlanRequest =>
    planRequestWithPlanId(transformationPlanRequest, transformationPlans)
  );
