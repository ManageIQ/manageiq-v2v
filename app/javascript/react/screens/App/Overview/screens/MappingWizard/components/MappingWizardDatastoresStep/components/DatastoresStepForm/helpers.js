import URI from 'urijs';

import {
  getTransformationMappingItemsBySourceType,
  getTransformationMappingItemsByDestinationType
} from '../../../../helpers';
import API from '../../../../../../../../../../common/API';
import {
  TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES,
  TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES
} from '../../../../MappingWizardConstants';
import { QUERY_ATTRIBUTES, FETCH_STORAGE_URLS, STORAGE_ATTRIBUTES } from '../../MappingWizardDatastoresStepConstants';

export const datastoreUsedSpace = (datastore = {}) => datastore.total_space - datastore.free_space;

export const totalUsedSpace = (datastores = []) =>
  datastores.reduce((totalSpace, datastore) => (totalSpace += datastoreUsedSpace(datastore)), 0);

export const targetDatastoreAvailableSpace = (targetDatastore, datastoresStepMappings) => {
  const datastoresMappings = datastoresStepMappings.reduce(
    (mappings, targetClusterWithDatastoresMappings) => mappings.concat(targetClusterWithDatastoresMappings.nodes),
    []
  );

  const matchingDatastoresMapping = datastoresMappings.find(
    targetDatastoreWithSourceDatastores => targetDatastoreWithSourceDatastores.id === targetDatastore.id
  );

  if (datastoresMappings.length > 0 && matchingDatastoresMapping) {
    return targetDatastore.free_space - totalUsedSpace(matchingDatastoresMapping.nodes);
  }
  return targetDatastore.free_space;
};

export const sourceDatastoreInfo = sourceDatastore =>
  sprintf(__('%s \\ %s \\ %s'), sourceDatastore.providerName, sourceDatastore.datacenterName, sourceDatastore.name);

export const targetDatastoreInfo = targetDatastore =>
  sprintf(__('%s \\ %s'), targetDatastore.providerName, targetDatastore.name);

export const targetDatastoreTreeViewInfo = targetDatastore => targetDatastore.name;

export const targetDatastoreWithTreeviewAttrs = targetDatastore => ({
  ...targetDatastore,
  text: targetDatastoreTreeViewInfo(targetDatastore),
  selectable: true,
  selected: false,
  state: {
    expanded: true
  }
});

export const sourceDatastoreWithTreeviewAttrs = sourceDatastore => ({
  ...sourceDatastore,
  text: sourceDatastore.name,
  icon: 'fa fa-file-o',
  selectable: true,
  selected: false
});

export const errorMessage = __('The size of the selected source datastores exceeds the available space in the target datastore'); // prettier-ignore

export const removeSourceDatastore = (datastoresMapping, nodeToRemove) => {
  const { nodes: sourceDatastores, ...targetDatastore } = datastoresMapping;
  const updatedSourceDatastores = sourceDatastores.filter(sourceDatastore => sourceDatastore.id !== nodeToRemove.id);
  return updatedSourceDatastores.length === 0
    ? undefined
    : {
        ...targetDatastore,
        nodes: updatedSourceDatastores
      };
};

export const updateMappings = (targetClusterWithDatastoresMappings, nodeToRemove) => {
  const isTargetDatastore = nodeToRemove.nodes;
  const { nodes: datastoresMappings, ...targetCluster } = targetClusterWithDatastoresMappings;

  const updatedDatastoresMappings = isTargetDatastore
    ? datastoresMappings.filter(
        targetDatastoreWithSourceDatastores => targetDatastoreWithSourceDatastores.id !== nodeToRemove.id
      )
    : datastoresMappings
        .map(datastoresMapping => removeSourceDatastore(datastoresMapping, nodeToRemove))
        .filter(item => item !== undefined);

  return updatedDatastoresMappings.length === 0
    ? undefined
    : {
        ...targetCluster,
        nodes: updatedDatastoresMappings
      };
};

const getMappedTargetDatastores = (datastores, mappedDatastoreIds) =>
  datastores.filter(datastore => mappedDatastoreIds.some(id => id === datastore.id));

const getMappedSourceDatastores = (targetDatastore, sourceDatastores, transformation) => {
  const datastoreMappingItems = getTransformationMappingItemsBySourceType(
    TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES.datastore,
    transformation
  );
  return sourceDatastores.filter(sourceDatastore =>
    datastoreMappingItems.some(
      item => item.destination_id === targetDatastore.id && item.source_id === sourceDatastore.id
    )
  );
};

const clusterWithStoragesUrl = (id, provider) => {
  const uri = new URI(`${FETCH_STORAGE_URLS[provider]}/${id}`);
  uri.addSearch({ attributes: QUERY_ATTRIBUTES[provider] });

  return uri.toString();
};

const targetDatastorePromises = (targetClusterIds, targetProvider) =>
  targetClusterIds.map(
    id =>
      new Promise((resolve, reject) => {
        API.get(clusterWithStoragesUrl(id, targetProvider))
          .then(res => {
            resolve(res.data);
          })
          .catch(e => reject(e));
      })
  );

const sourceDatastorePromises = sourceClusterIds =>
  sourceClusterIds.map(
    id =>
      new Promise((resolve, reject) => {
        API.get(clusterWithStoragesUrl(id, 'source'))
          .then(res => {
            resolve(res.data.storages.map(storage => ({ ...storage, sourceClusterId: id })));
          })
          .catch(e => reject(e));
      })
  );

export const createDatastoresMappings = (transformation, targetProvider) =>
  new Promise((resolve, reject) => {
    const targetClusterIds = getTransformationMappingItemsByDestinationType(
      TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES[targetProvider].cluster,
      transformation
    ).map(item => item.destination_id);
    const sourceClusterIds = getTransformationMappingItemsBySourceType(
      TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES.cluster,
      transformation
    ).map(item => item.source_id);
    const targetDatastoreIds = getTransformationMappingItemsByDestinationType(
      TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES[targetProvider].datastore,
      transformation
    ).map(item => item.destination_id);

    Promise.all([
      ...targetDatastorePromises(targetClusterIds, targetProvider),
      ...sourceDatastorePromises(sourceClusterIds)
    ])
      .then(res => {
        const datastoresMappings = [];
        const targetClustersWithDatastores = res.filter(resolvedPromise => !resolvedPromise.length);
        const sourceDatastores = res.reduce((datastores, resolvedPromise) => {
          if (resolvedPromise.length) {
            return [...datastores, ...resolvedPromise];
          }
          return datastores;
        }, []);
        targetClustersWithDatastores.forEach(mapping => {
          const nodes = getMappedTargetDatastores(mapping[STORAGE_ATTRIBUTES[targetProvider]], targetDatastoreIds).map(
            targetDatastore => ({
              ...targetDatastoreWithTreeviewAttrs(targetDatastore),
              nodes: getMappedSourceDatastores(targetDatastore, sourceDatastores, transformation).map(datastore =>
                sourceDatastoreWithTreeviewAttrs(datastore)
              )
            })
          );

          datastoresMappings.push({
            ...mapping,
            text: mapping.name,
            selectable: false,
            state: {
              expanded: true
            },
            nodes
          });
        });

        resolve(datastoresMappings);
      })
      .catch(e => reject(e));
  });
