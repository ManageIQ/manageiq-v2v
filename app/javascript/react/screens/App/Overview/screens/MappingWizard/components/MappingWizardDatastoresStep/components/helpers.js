import numeral from 'numeral';

export const datastoreUsedSpace = (datastore = {}) =>
  datastore.total_space - datastore.free_space;

export const totalUsedSpace = (datastores = []) =>
  datastores.reduce(
    (totalSpace, datastore) => (totalSpace += datastoreUsedSpace(datastore)),
    0
  );

export const targetDatastoreAvailableSpace = (
  targetDatastore,
  datastoresStepMappings
) => {
  const datastoresMappings = datastoresStepMappings.reduce(
    (mappings, targetClusterWithDatastoresMappings) =>
      mappings.concat(targetClusterWithDatastoresMappings.nodes),
    []
  );

  const matchingDatastoresMapping = datastoresMappings.find(
    targetDatastoreWithSourceDatastores =>
      targetDatastoreWithSourceDatastores.id === targetDatastore.id
  );

  if (datastoresMappings.length > 0 && matchingDatastoresMapping) {
    return (
      targetDatastore.free_space -
      totalUsedSpace(matchingDatastoresMapping.nodes)
    );
  }
  return targetDatastore.free_space;
};

export const sourceDatastoreInfo = sourceDatastore =>
  sprintf(
    __('%s (%s)'),
    sourceDatastore.name,
    numeral(datastoreUsedSpace(sourceDatastore)).format('0.00b')
  );

export const targetDatastoreInfo = (targetDatastore, datastoresStepMappings) =>
  sprintf(
    __('%s (%s avail)'),
    targetDatastore.name,
    numeral(
      targetDatastoreAvailableSpace(targetDatastore, datastoresStepMappings)
    ).format('0.00b')
  );

export const targetDatastoreTreeViewInfo = (
  targetDatastore,
  sourceDatastores
) => {
  const { total_space, free_space } = targetDatastore;
  const availableSpace = free_space - totalUsedSpace(sourceDatastores);

  return sprintf(
    __('%s (%s total, %s avail)'),
    targetDatastore.name,
    numeral(total_space).format('0.00b'),
    numeral(availableSpace).format('0.00b')
  );
};

export const errorMessage = __(
  'The size of the selected source datastores exceeds the available space in the target datastore'
);
