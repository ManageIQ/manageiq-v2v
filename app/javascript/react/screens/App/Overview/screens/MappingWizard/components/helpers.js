import { V2V_TARGET_PROVIDERS } from '../MappingWizardConstants';

export const getClusterOptions = clusterMappings => {
  const sourceClustersWithAssociatedTargetClusters = clusterMappings.reduce(
    (mappings, targetClusterWithSourceClusters) => {
      const { nodes: sourceClusters, ...targetCluster } = targetClusterWithSourceClusters;
      const sourceToTargetMappings = sourceClusters.map(sourceCluster => ({
        sourceCluster,
        targetCluster,
        sourceClusterMappedToTargetCluster: {
          name: `${sourceCluster.name} (${targetCluster.name})`,
          id: sourceCluster.id
        }
      }));
      return mappings.concat(sourceToTargetMappings);
    },
    []
  );

  return sourceClustersWithAssociatedTargetClusters.map(
    ({ sourceClusterMappedToTargetCluster }) => sourceClusterMappedToTargetCluster
  );
};

export const multiProviderTargetLabel = (providerType, wizardStep) => {
  switch (providerType) {
    case V2V_TARGET_PROVIDERS[1].id:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Project');
        case 'storage':
          return __('Target Provider \\ Volume Type');
        case 'network':
          return __('Target Project \\ Network');
        default:
          return null;
      }
    default:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Datacenter \\ Cluster');
        case 'storage':
          return __('Target Datastores');
        case 'network':
          return __('Target Network');
        default:
          return null;
      }
  }
};

export const updateMappings = (mappings, clusterStepMappings) =>
  mappings.filter(mapping => {
    const potentialAssociatedClusterStepMappings = clusterStepMappings.filter(
      clusterStepMapping => clusterStepMapping.id === mapping.id
    );

    if (potentialAssociatedClusterStepMappings.length === 0) {
      return false;
    }

    const associatedClusterMapping = potentialAssociatedClusterStepMappings.find(clusterStepMapping => {
      const { nodes: sourceClusters } = clusterStepMapping;
      const sourceClusterIds = sourceClusters.map(sourceCluster => sourceCluster.id);
      const { nodes: mappedSourceStoragesOrNetworks } = mapping.nodes[0];

      return mappedSourceStoragesOrNetworks.every(storageOrNetwork =>
        sourceClusterIds.some(id => id === storageOrNetwork.sourceClusterId)
      );
    });

    return associatedClusterMapping;
  });
