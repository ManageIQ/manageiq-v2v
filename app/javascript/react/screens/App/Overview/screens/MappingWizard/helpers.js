export const createTransformationMappings = (
  mappingWizardGeneralStep,
  mappingWizardClustersStep,
  mappingWizardDatastoresStep,
  mappingWizardNetworksStep
) => {
  const clustersUrlRegEx = /\/api\/clusters\/\d{1,}/;
  const clusterTransformationMappings = mappingWizardClustersStep.values.clusterMappings.reduce(
    (clusterTransformationsArray, targetClusterWithSourceClusters) => {
      const destination = targetClusterWithSourceClusters.href.match(
        clustersUrlRegEx
      )[0];
      const transformations = targetClusterWithSourceClusters.nodes.reduce(
        (clusterTransformations, sourceCluster) =>
          clusterTransformations.concat({
            source: sourceCluster.href.match(clustersUrlRegEx)[0],
            destination
          }),
        []
      );
      return clusterTransformationsArray.concat(transformations);
    },
    []
  );

  const datastoresUrlRegEx = /\/api\/data_stores\/\d{1,}/;
  const datastoreTransformationMappings = mappingWizardDatastoresStep.values.datastoresMappings.reduce(
    (
      datastoreTransformationsPerTargetCluster,
      targetClusterWithDatastoreMappings
    ) => {
      const datastoreTransformationsForTargetCluster = targetClusterWithDatastoreMappings.nodes.reduce(
        (
          datastoreTransformationsPerDatastoreMapping,
          targetDatastoreWithSourceDatastores
        ) => {
          const datastoreTransformations = targetDatastoreWithSourceDatastores.nodes.reduce(
            (transformations, sourceDatastore) =>
              transformations.concat({
                source: sourceDatastore.href.match(datastoresUrlRegEx)[0],
                destination: targetDatastoreWithSourceDatastores.href.match(
                  datastoresUrlRegEx
                )[0]
              }),
            []
          );
          return datastoreTransformationsPerDatastoreMapping.concat(
            datastoreTransformations
          );
        },
        []
      );
      return datastoreTransformationsPerTargetCluster.concat(
        datastoreTransformationsForTargetCluster
      );
    },
    []
  );

  const networksUrlRegEx = /\/api\/lans\/\d{1,}/;
  const networkTransformationMappings = mappingWizardNetworksStep.values.networksMappings.reduce(
    (
      networkTransformationsPerTargetCluster,
      targetClusterWithNetworkMappings
    ) => {
      const networkTransformationsForTargetCluster = targetClusterWithNetworkMappings.nodes.reduce(
        (
          networkTransformationsPerNetworkMapping,
          targetNetworkWithSourceNetworks
        ) => {
          const networkTransformations = targetNetworkWithSourceNetworks.nodes.reduce(
            (transformations, sourceNetwork) =>
              transformations.concat({
                source: sourceNetwork.href.match(networksUrlRegEx)[0],
                destination: targetNetworkWithSourceNetworks.href.match(
                  networksUrlRegEx
                )[0]
              }),
            []
          );
          return networkTransformationsPerNetworkMapping.concat(
            networkTransformations
          );
        },
        []
      );
      return networkTransformationsPerTargetCluster.concat(
        networkTransformationsForTargetCluster
      );
    },
    []
  );

  return {
    name: mappingWizardGeneralStep.values.name,
    description: mappingWizardGeneralStep.values.description,
    state: 'draft',
    transformation_mapping_items: [
      ...clusterTransformationMappings,
      ...datastoreTransformationMappings,
      ...networkTransformationMappings
    ]
  };
};
