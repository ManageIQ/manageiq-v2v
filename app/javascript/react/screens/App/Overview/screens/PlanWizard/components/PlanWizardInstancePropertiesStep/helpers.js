import { OSP_TENANT, EMS_CLUSTER } from '../../../../OverviewConstants';

export const getTenantsById = tenants =>
  tenants.reduce(
    (tenantsById, tenant) => ({
      ...tenantsById,
      [tenant.id]: { ...tenant }
    }),
    {}
  );

export const getDestinationTenantIdsBySourceClusterId = transformation_mapping_items => {
  const relevantMappings = transformation_mapping_items.filter(
    item => item.destination_type === OSP_TENANT && item.source_type === EMS_CLUSTER
  );
  return relevantMappings.reduce(
    (newObject, mapping) => ({
      ...newObject,
      [mapping.source_id]: mapping.destination_id
    }),
    {}
  );
};

export const getVmsWithTargetClusterName = (vms, destinationTenantIdsBySourceClusterId, tenantsWithAttributesById) =>
  vms.map(vm => {
    const destinationTenantId = destinationTenantIdsBySourceClusterId[vm.ems_cluster_id];
    const tenant = destinationTenantId && tenantsWithAttributesById[destinationTenantId];
    return {
      ...vm,
      target_cluster_name: tenant ? tenant.name : ''
    };
  });

export const allFitForVM = (bestFitFlavors, tenantFlavors, vmId) => {
  const flavorsSlugPrefix = 'flavors/';
  const flavorForVM = bestFitFlavors.find(flavor => flavor.source_href_slug === `vms/${vmId}`);
  const allFitFlavors = flavorForVM && flavorForVM.all_fit;

  if (!allFitFlavors || allFitFlavors.length === 0) {
    return [tenantFlavors.reduce((prev, current) => (prev.root_disk_size > current.root_disk_size ? prev : current))];
  }
  const allFitFlavorIds = allFitFlavors.map(flavorSlug => flavorSlug.slice(flavorsSlugPrefix.length));
  return tenantFlavors.filter(flavor => allFitFlavorIds.indexOf(flavor.id) > -1);
};
