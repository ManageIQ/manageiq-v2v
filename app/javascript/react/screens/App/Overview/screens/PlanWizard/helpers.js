export const createMigrationPlans = (
  planWizardGeneralStep,
  planWizardVMStep
) => {
  const planName = planWizardGeneralStep.values.name;
  const planDescription = planWizardGeneralStep.values.description;
  const infrastructureMapping =
    planWizardGeneralStep.values.infrastructure_mapping;
  const vms = planWizardVMStep.values.selectedVms;

  return {
    name: planName,
    description: planDescription,
    prov_type: 'generic_transformation_plan',
    config_info: {
      transformation_mapping_id: infrastructureMapping,
      vm_ids: vms
    }
  };
};
