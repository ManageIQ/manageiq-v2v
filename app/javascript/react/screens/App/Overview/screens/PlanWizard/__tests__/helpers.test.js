import { createMigrationPlans } from '../helpers';
import { ospPlanReduxFormState, rhvPlanReduxFormState } from '../planWizard.fixtures';

describe('createMigrationPlans', () => {
  test('with an OSP target', () => {
    const {
      planWizardGeneralStep,
      planWizardVMStep,
      planWizardInstancePropertiesStep,
      planWizardAdvancedOptionsStep,
      planWizardScheduleStep
    } = ospPlanReduxFormState;
    const plansBody = createMigrationPlans(
      planWizardGeneralStep,
      planWizardVMStep,
      planWizardInstancePropertiesStep,
      planWizardAdvancedOptionsStep,
      planWizardScheduleStep
    );
    expect(plansBody).toMatchSnapshot();
  });

  test('with a RHV target and playbooks', () => {
    const {
      planWizardGeneralStep,
      planWizardVMStep,
      planWizardInstancePropertiesStep,
      planWizardAdvancedOptionsStep,
      planWizardScheduleStep
    } = rhvPlanReduxFormState;
    const plansBody = createMigrationPlans(
      planWizardGeneralStep,
      planWizardVMStep,
      planWizardInstancePropertiesStep,
      planWizardAdvancedOptionsStep,
      planWizardScheduleStep
    );
    expect(plansBody).toMatchSnapshot();
  });
});
