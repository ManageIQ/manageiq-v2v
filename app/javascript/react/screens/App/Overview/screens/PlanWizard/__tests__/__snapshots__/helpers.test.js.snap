// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMigrationPlans with a RHV target and playbooks 1`] = `
Object {
  "config_info": Object {
    "actions": Array [
      Object {
        "osp_flavor_id": undefined,
        "osp_security_group_id": undefined,
        "post_service": false,
        "pre_service": true,
        "vm_id": "28",
      },
      Object {
        "osp_flavor_id": undefined,
        "osp_security_group_id": undefined,
        "post_service": true,
        "pre_service": false,
        "vm_id": "36",
      },
    ],
    "post_service_id": "2",
    "pre_service_id": "1",
    "transformation_mapping_id": "1",
    "warm_migration": false,
  },
  "description": "",
  "name": "test RHV w/ playbooks",
  "prov_type": "generic_transformation_plan",
}
`;

exports[`createMigrationPlans with an OSP target 1`] = `
Object {
  "config_info": Object {
    "actions": Array [
      Object {
        "osp_flavor_id": "42000000000004",
        "osp_security_group_id": "42000000000016",
        "post_service": false,
        "pre_service": false,
        "vm_id": "42000000000017",
      },
      Object {
        "osp_flavor_id": "42000000000002",
        "osp_security_group_id": "42000000000022",
        "post_service": false,
        "pre_service": false,
        "vm_id": "42000000000006",
      },
    ],
    "post_service_id": "",
    "pre_service_id": "",
    "transformation_mapping_id": "42000000000002",
    "warm_migration": false,
  },
  "description": "",
  "name": "test OSP plan",
  "prov_type": "generic_transformation_plan",
}
`;
